library/Rules/NoWhitespace.php

```diff
@@ -5,6 +5,10 @@ class NoWhitespace extends AbstractRule
 {
     public function validate($input)
     {
+        if (is_array($input)) {
```

Comment:
I guess the `is_null()` below is there to cover optional input only but let's keep it in order to avoid BC breaks.
The `is_scalar()` function does not consider NULL to be scalar.

Even with this fix the rule will fail when raise errors with non-scalar values.